django_components/component.py

```diff
@@ -41,17 +42,21 @@ def __new__(mcs, name, bases, attrs):
 
 
 class Component(metaclass=SimplifiedInterfaceMediaDefiningClass):
+    template_name = None
 
     def __init__(self, component_name):
         self._component_name = component_name
         self.instance_template = None
         self.slots = {}
 
-    def context(self):
-        return {}
+    def get_context(self, *args, **kwargs):
+        return kwargs
```

Comment:
It's done to be able to pass any `kwargs` to component template without overriding `get_context` for every component. In Django context data is not validated in any way, here's code from `ContextMixin` (a "parent" of all views with context):
```python
    def get_context_data(self, **kwargs):
        kwargs.setdefault('view', self)
        if self.extra_context is not None:
            kwargs.update(self.extra_context)
        return kwargs
```
I usually override `get_context_data` this way:
```python
    def get_context_data(self, **kwargs):
        kwargs['some_model_form'] = SomeModelForm(instance=self.object, readonly=True, user=self.request.user)
        return super().get_context_data(**kwargs)
```
You can throw any dict to it and it'll be available in template.
By the way now I see I made a mistake by calling the method `get_context` it shall be `get_context_data` :D

Reply:
I see the thinking here. The problem with this is that looking at the component, there's no way to know what parameters it takes. So the only way to find that out is to look through the template, and try to find every parameter.

I don't think we should make this change, because I really like the documentation that always overrriding get_context_data provides.

Reply:
@Real-Gecko Hi! Let me know if you want help or if you disagree and want to talk more about this.

Reply:
Sorry for late reply. The idea behind Django templates is that they [fail silently if variable is not found](https://docs.djangoproject.com/en/3.0/ref/templates/api/#invalid-template-variables), that's by design. So I disagree with enforcing `get_context_data` override. This way components become no different from [inclusion tags](https://docs.djangoproject.com/en/3.2/howto/custom-template-tags/#howto-custom-template-tags-inclusion-tags) and it's not that exiting to enumerate all possible parameters with something like `return {'choices': choices,...}`. If someone wants to ensure required parameters he can do so freely and override parent method, but for parent not having this method implemented is a flaw IMO. So this way is better:
```python
    def get_context(self, **kwargs):
        return kwargs
```

Reply:
I recognize that django templates fail if a variable is not found, and I think that was a horrible decision. It's lead to several productions bugs at work. I've seen core committer agreeing but decided against it because large parts of the admin mode depends on it. 

The core problem is that looking at call sites to understand how something works is too messy. 

To illustrate: Say you working on adding a calendar to a part of a big django project you're new working on. You know that the project uses django-components, so you look in the components.py file. You find a calendar component, great! Now, how do I use it? If get_context_data is defined you now have everything you need. If not, you're off searching for "{% component 'calendar" (or should that single quote be a double quote? Yeah, and component_block too right?) across the project, trying to find someone that uses the calendar as you want. Or you could dive into the template file and reverse the parameters from there. If the template is small that's doable, if it's big, not so much.

I see the downsides of having to repeat yourself, just like when sending context to render() at the end of a view, but I think this is and OK trade-off.

Reply:
Well TBH I really tired of generating HTML with Django serverside, even with all pros and cons of Django template system and variety of apps available(like this one). And I guess I'm not alone here, that's why frameworks like React were created I guess. But alas I cannot switch to React in my current project. So I'm looking for things that make life easier on Django side.
And having `get_context_data` like this is better IMO
```python
    def get_context_data(self, **kwargs):
        return kwargs
```

Reply:
I don't agree, for the reasons above. But, since this is just inheritance, it's easy to just have a AutoContextComponent in a project that behaves like you want it too, and just inherit from that one instead.

Reply:
No, I try to avoid that, because project already has enough of monkey patches. For example caused by this one https://github.com/carltongibson/django-filter/pull/1268

Reply:
That request has nothing to do with this patch. The similarity is that it's "also annoying for you". I'm open to other arguments.